R/GeospatialFunctions.R

```diff
@@ -0,0 +1,571 @@
+#' @return The original `TADA_DataRetrieval()` dataframe with additional columns associated with the ATTAINS assessment unit data. 
+#' 
+#' @seealso [TADA_DataRetrieval()]
+#' @seealso [TADA_MakeSpatial()]
+#' @seealso [TADA_ViewATTAINS()]
+#' 
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#'tada_data <- TADA_DataRetrieval(startDate = "1990-01-01",
+#'                                endDate = "1995-12-31",
+#'                                characteristicName = "pH",
+#'                                statecode = "NV",
+#'                                applyautoclean = TRUE)
+#'                                  
+#'tada_attains <- TADA_GetATTAINS(data = tada_data, return = TRUE)
+#' }
+
+TADA_GetATTAINS <- function(data, return = FALSE){
+  
+  if(nrow(data) == 0){
+    
+    print("Your Water Quality Portal dataframe has no observations. Returning an empty dataframe.")
+    
+    # if no WQP observations, return a modified `data` with empty ATTAINS-related columns:
+    no_WQP_data <- data %>%
+      dplyr::mutate("ATTAINS.organizationid" = NA, "ATTAINS.submissionid" = NA, "ATTAINS.hasprotectionplan" = NA,
+                    "ATTAINS.assessmentunitname" = NA, "ATTAINS.nhdplusid" = NA, "ATTAINS.tas303d" = NA,                                                  
+                    "ATTAINS.isthreatened" = NA, "ATTAINS.state" = NA, "ATTAINS.on303dlist" = NA,                                               
+                    "ATTAINS.organizationname" = NA, "ATTAINS.region" = NA, "ATTAINS.Shape_Length" = NA,                                             
+                    "ATTAINS.reportingcycle" = NA, "ATTAINS.assmnt_joinkey" = NA, "ATTAINS.hastmdl" = NA,                                                  
+                    "ATTAINS.orgtype" = NA, "ATTAINS.permid_joinkey" = NA, "ATTAINS.catchmentistribal" = NA,                                        
+                    "ATTAINS.ircategory" = NA, "ATTAINS.waterbodyreportlink" = NA, "ATTAINS.assessmentunitidentifier" = NA,                                 
+                    "ATTAINS.overallstatus" = NA, "ATTAINS.isassessed" = NA, "ATTAINS.isimpaired" = NA,                                               
+                    "ATTAINS.has4bplan" = NA, "ATTAINS.huc12" = NA, "ATTAINS.hasalternativeplan" = NA,                                      
+                    "ATTAINS.visionpriority303d" = NA, "ATTAINS.areasqkm" = NA, "ATTAINS.catchmentareasqkm" = NA,                                       
+                    "ATTAINS.catchmentstatecode" = NA, "ATTAINS.catchmentresolution" = NA, "ATTAINS.Shape_Area" = NA)
+    
+    return(no_WQP_data)
+    
+  }
+  
+  suppressMessages(suppressWarnings({
+    
+    sf::sf_use_s2(FALSE)
+    
+    # If data is already spatial, just make sure it is in the right CRS
+    # and add an index as the WQP observations' unique IDs...
+    if (!is.null(data) & inherits(data, "sf")) {
+      TADA_DataRetrieval_data <- data %>%
+        sf::st_transform(4326) %>%
```

Comment:
For the functions that check for {sf} format, it seems like it would be more efficient to just check for 4326 and only transform if needed? Especially if they're large?